shell.nix

```diff
@@ -6,6 +6,16 @@ let
   pyEnv = pkgs.python3.withPackages (ps: [ packageSet.sphinxcontrib-haddock.sphinxcontrib-domaintools ps.sphinx ps.sphinx_rtd_theme ]);
   # Called from Cabal to generate the Haskell source for the metatheory package
   agdaWithStdlib = agdaPackages.agda.withPackages [ agdaPackages.standard-library ];
+  # Configure project pre-commit hooks
+  pre-commit-check = pkgs.nix-pre-commit-hooks.run {
+    src = ./.;
```

Comment:
Should this have a filter?

Reply:
I could turn this into `src = (pkgs.lib.cleanSource ./.)` ? 

Or are you thinking of a more specific inclusion list? I'm afraid this might become annoying to keep up to date? Considering also that this isn't restricted to Haskell files only necessarily (even though it is *right now*).

Reply:
Yeah, that might do it. I'm just wary of bare `src = ./.` as I've been bitten by its poor caching behaviour too many times!